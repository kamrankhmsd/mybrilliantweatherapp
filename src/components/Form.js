import React, { Component } from 'react';
import Locations from './Locations';
import Header from './Header';
import '../App.css'

class Form extends Component {
    constructor(props) {
        super(props);
        this.state = {
            locations: [],
            searchItem: '',
            message: ''
        };
    }

    getLocations = (e) => {
        e.preventDefault();
        let apiQuery = this.state.searchItem;
        this.setState({ searchItem: '' });
        if (apiQuery) {
            this.setState({ message: 'LOADING...' });
            fetch(`https://cors-anywhere.herokuapp.com/https://www.metaweather.com/api/location/search/?query=${apiQuery}`)
                .then(result => result.json())
                .then(location => {
                    let locationsArray = location.slice();
                    { locationsArray.length > 0 ? this.setState({ locations: locationsArray, message: 'SEARCH RESULTS' }) : this.setState({ locations: locationsArray, message: 'LOCATION NOT FOUND' }) };
                }).catch(err => {
                    console.log(err);
                    this.setState({ message: 'ERROR: REQUEST FAILED' });
                });
        } else {
            this.setState({ message: 'PLEASE ENTER A LOCATION' });
        }
    }

    onChange = (e) => {
        this.setState({ searchItem: e.target.value });
    }

    render() {
        return (
            <div>
                <Header />
                <div className="wrapper">
                    <div className="form">
                        <form className="form" onSubmit={this.getLocations}>
                            <input className="location-search-input" type="text" onChange={this.onChange} placeholder="location..." autoFocus value={this.state.searchItem} />
                            <button className="location-search-button">Find locations</button>
                        </form>
                        <Locations locations={this.state.locations} message={this.state.message} />
                    </div>
                </div>
            </div>
        );
    };
}

export default Form;