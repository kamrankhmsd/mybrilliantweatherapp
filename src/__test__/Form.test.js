import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Form from '../components/Form';

Enzyme.configure({ adapter: new Adapter() });


describe('Form Component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Form />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});



