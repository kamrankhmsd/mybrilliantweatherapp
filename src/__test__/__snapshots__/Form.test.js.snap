// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component matches the snapshot 1`] = `
<div>
  <div
    className="header"
  >
    <h1>
      My Brilliant Weather App
    </h1>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="form"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          autoFocus={true}
          className="location-search-input"
          onChange={[Function]}
          placeholder="location..."
          type="text"
          value=""
        />
        <button
          className="location-search-button"
        >
          Find locations
        </button>
      </form>
      <div>
        <span>
          
        </span>
        <div />
      </div>
    </div>
  </div>
</div>
`;
